cpp/CMakeLists.txt

```diff
@@ -949,10 +949,9 @@ add_subdirectory(src/arrow)
 add_subdirectory(src/arrow/io)
 if (ARROW_IPC)
   add_subdirectory(src/arrow/ipc)
+  set(ARROW_DEPENDENCIES ${ARROW_DEPENDENCIES} metadata_fbs)
```

Comment:
The dependency target "metadata_fbs" of target "arrow_objlib" does not
    exist.
  Call Stack (most recent call first):
    CMakeLists.txt:972 (ADD_ARROW_LIB)

dependency to the metadata_fbs target is always set , but it defined conditionally:
if (ARROW_IPC)
  add_subdirectory(src/arrow/ipc)